components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -162,59 +162,59 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
-      this.team = team;
-      this.updateNameForm.controls.name.setValue(this.team.name);
-      this.store.dispatch(new GetTeamUsers({ id: this.teamId }));
-      this.store.dispatch(new GetUsers());
-      if (this.projectsEnabled) {
-        this.store.dispatch(new GetProjects());
-      }
-    });
+        this.team = team;
+        this.updateNameForm.controls.name.setValue(this.team.name);
+        this.store.dispatch(new GetTeamUsers({ id: this.teamId }));
+        this.store.dispatch(new GetUsers());
+        if (this.projectsEnabled) {
+          this.store.dispatch(new GetProjects());
+        }
+      });
 
     combineLatest([
       this.store.select(allProjects),
       this.store.select(getAllProjectStatus)
     ]).pipe(
       takeUntil(this.isDestroyed),
       filter(([_, pStatus]: [Project[], EntityStatus]) => pStatus !== EntityStatus.loading),
-      filter(() => !!this.team),
-      map(([allowedProjects, _]) => {
+      filter(() => !!this.team)
+    ).subscribe(([allowedProjects, _]) => {
         this.projects = {};
         allowedProjects
           .forEach(p => {
             this.projects[p.id] = { ...p, checked: this.team.projects.includes(p.id)
             };
           });
-      }))
-      .subscribe();
+      });
 
-    this.sortedUsers$ = <Observable<User[]>>combineLatest([
+    combineLatest([
       this.store.select(allUsers),
       this.store.select(userStatus),
       this.store.select(teamUsers),
-      this.store.select(getUsersStatus)])
-      .pipe(
-        map(([users, uStatus, tUsers, tStatus]: [User[], EntityStatus, string[], EntityStatus]) => {
-          if (uStatus !== EntityStatus.loadingSuccess ||
-            tStatus !== EntityStatus.loadingSuccess) {
-            return [];
-          }
-          // Map UUID membership to user records and remove any entries that don't
-          // map to user records.
-          return at(tUsers, keyBy('membership_id', users))
-            .filter(userRecord => userRecord !== undefined);
+      this.store.select(getUsersStatus)]).pipe(
+        takeUntil(this.isDestroyed),
+        filter(([_allUsers, uStatus, _teamUsers, tuStatus]:
+          [User[], EntityStatus, string[], EntityStatus]) => {
+            return uStatus === EntityStatus.loadingSuccess ||
+              tuStatus === EntityStatus.loadingSuccess;
         }),
-        map((users: User[]) => users.sort(
-          (a, b) => {
-            // See https://stackoverflow.com/a/38641281 for these options
-            const opts = { numeric: true, sensitivity: 'base' };
-            // sort by name then by id
-            return a.name.localeCompare(b.name, undefined, opts) ||
-              a.name.localeCompare(b.name, undefined, { numeric: true }) ||
-              a.id.localeCompare(b.id, undefined, opts);
-            })),
-        takeUntil(this.isDestroyed)
-      );
+        filter(() => !!this.team),
+        map(([users, _uStatus, teamUserIds, _tuStatus]:
```

Comment:
Can also be simplified: if the `=> { ... }` is `=> { return x }`, we can replace the thing by just `=> x`.

Reply:
Ok to leave it? I'm new-ish to typescript and returning that multiline function (as opposed to a nice lil var like `x`) harder to visually parse...

Reply:
Sure! 😃